packages/dicom/dcmtk/write-segmentation.cxx

```diff
@@ -0,0 +1,176 @@
+/*=========================================================================
+
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+// DCMQI includes
+#include "dcmqi/Itk2DicomConverter.h"
+#undef HAVE_SSTREAM // Avoid redefinition warning
+#include "dcmqi/internal/VersionConfigure.h"
+
+// DCMTK includes
+#include "dcmtk/oflog/configrt.h"
+
+#include "itkPipeline.h"
+#include "itkInputImage.h"
+#include "itkOutputBinaryStream.h"
+#include "itkSupportInputImageTypes.h"
+#include "itkSmartPointer.h"
+#include "itkInputTextStream.h"
+
+typedef dcmqi::Helper helper;
+constexpr unsigned int Dimension = 3;
+using PixelType = short;
+using ScalarImageType = itk::Image<PixelType, Dimension>;
+
+int runPipeline(
+  itk::wasm::Pipeline & pipeline,
+  itk::wasm::InputImage<ScalarImageType>& inputImage,
+  itk::wasm::InputTextStream& metaInfoStream,
```

Comment:
const?

Reply:
Might need to add a .Get() const function to InputTextStream class. I will run an experiment to see if it builds fine.

Reply:
I think this one is a little more complicated than I thought. We should leave it as non-const for now.
The `std::istream` object possibly self-modifies itself when its being read which is why there's no const way to convert it into an `std::string`.